packages/api/src/on/on.ts

```diff
@@ -0,0 +1,18 @@
+import * as Sentry from '@sentry/react';
+
+import { EventEventListener } from '@gemwallet/constants';
+
+export const on = (eventType: string, callback: (payload: any) => void) => {
+  window.addEventListener('message', (event: EventEventListener) => {
+    try {
+      // We only accept messages from ourselves
+      if (event.source !== window) return;
```

Comment:
@FlorianBouron Let's check together the controls that need to be put here

Reply:
You could pass additional data within the events, to make sure it comes from GemWallet, eg: https://github.com/GemWallet/gemwallet-extension/blob/master/packages/api/src/helpers/extensionMessaging.ts#L58

Reply:
I see that you actually pass the `app` and `source`:
https://github.com/GemWallet/gemwallet-extension/pull/193/files#diff-d6ff6fb3a7f713e467cca8535e399062665b54bc45ad0d4e8fd913a980b1ebfbR305
https://github.com/GemWallet/gemwallet-extension/pull/193/files#diff-d6ff6fb3a7f713e467cca8535e399062665b54bc45ad0d4e8fd913a980b1ebfbR307
These could be used for a checking.

Reply:
done